sdk/servicebus/azure-messaging-servicebus/README.md

```diff
@@ -434,14 +434,24 @@ try {
 }
 ```
 ### When to use 'ServiceBusProcessorClient'.
- When to use 'ServiceBusProcessorClient', 'ServiceBusReceiverClient' or ServiceBusReceiverAsyncClient? The processor 
- is built using 'ServiceBusReceiverAsyncClient', it provides a convenient way of receiving messages with default 
- auto complete and auto-renew of message locks in 'PEEK_LOCK' mode. The processor is appropriate where the 
- applications have not made complete move to async receiver client and want to process message in synchronous mode. 
- The processor receives messages forever because it recovers from the network errors internally. 
- 'ServiceBusProcessorClient:processMessage()' function call is made for each message. Alternatively, You can also use 
- 'ServiceBusReceiverClient', it is a lower level client and provides a wider range of APIs. If async processing is  
- suitable for your application, you can use 'ServiceBusReceiverAsyncClient'. 
+
+When to use 'ServiceBusProcessorClient', 'ServiceBusReceiverClient' or ServiceBusReceiverAsyncClient? 
+ 
+For almost all common use cases, 'ServiceBusProcessorClient' should be the primary choice for consuming messages. The processor is built with 
+necessary resiliency to recover from errors and designed to receive messages forever. The processor offers simplified API to configure 
+concurrency for asynchronous processing.
+
+Use of the low-level client, 'ServiceBusReceiverAsyncClient', expects proficiency in Reactive programming and must be used only when building 
+a Reactive library or an end-to-end Reactive application. Unlike processor, the low-level 'ServiceBusReceiverAsyncClient' doesn’t have resiliency built into it. 
+The owning (reactive) application is responsible to handle terminal events and pick the operators (to incorporate resiliency, manage backpressure, threading) 
+in the Reactor chain.  As stated earlier, for common asynchronous receive use cases, the 'ServiceBusProcessorClient' must be the primary choice.
+
+Use 'ServiceBusReceiverClient', if it is desired to receive messages synchronously, typical use case will have a loop, that calls `receiveMessages`on 
+'ServiceBusReceiverClient' instance, process the message(s) and the loop repeat this serial receive pattern. For non-serial high
```

Comment:
by non-serial, do you mean parallel?

Is there any benefit in using sync receiver vs processor with concurrency set to 1?
I guess processor would be better in all cases (resiliency, auto-lock renewals, observability).

Essentially, can we say that there is no use case where sync receiver client is necessary except maybe if you come from track-1 SDK where it was used and want to migrate with as little changes as possible?

Reply:
> by non-serial, do you mean parallel?

yes, how about "_For high-throughput and parallel message processing use cases, use 'ServiceBusProcessorClient_"

Reply:
> Is there any benefit in using sync receiver vs processor with concurrency set to 1? I guess processor would be better in all cases (resiliency, auto-lock renewals, observability).

Yes, the processor is useful in most cases. One case we found somebody wants to use SyncReceiver is - if they don’t want SDK to push messages, but needs to poll service bus in response to some event.  As in, some other component push messages to service bus, sent a notification to the receiving application, and upon receiving this notification only it will pull message from the service, then wait for next notification.